controllers/vaultstaticsecret_controller.go

```diff
@@ -218,11 +245,242 @@ func (r *VaultStaticSecretReconciler) handleDeletion(ctx context.Context, o clie
 	return nil
 }
 
+func (r *VaultStaticSecretReconciler) ensureEventWatcher(ctx context.Context, o *secretsv1beta1.VaultStaticSecret, c vault.Client) error {
+	logger := log.FromContext(ctx)
+	name := types.NamespacedName{Namespace: o.Namespace, Name: o.Name}
+
+	meta, ok := r.eventWatcherRegistry.Get(name)
+	if ok {
+		// The watcher is running, and if the VSS object has not been updated,
+		// and the client ID is the same, just return
+		if meta.LastGeneration == o.GetGeneration() && meta.LastClientID == c.ID() {
+			logger.V(consts.LogLevelDebug).Info("Event watcher already running",
+				"namespace", o.Namespace, "name", o.Name)
+			return nil
+		}
+	}
+	if meta != nil {
+		// The watcher is running, but the metadata or vault client has changed,
+		// so kill it
+		if meta.Cancel != nil {
+			meta.Cancel()
+			// Wait for the goroutine to stop and remove itself from the event registry
+			<-meta.StoppedCh
+		} else {
+			logger.Error(fmt.Errorf("nil cancel function"), "event watcher has nil cancel function", "meta", meta)
+		}
+	}
+	wsClient, err := c.WebsocketClient(eventPath)
+	if err != nil {
+		return fmt.Errorf("failed to create websocket client: %w", err)
+	}
+
+	watchCtx, cancel := context.WithCancel(context.Background())
+	stoppedCh := make(chan struct{})
+	updatedMeta := &eventWatcherMeta{
+		Cancel:         cancel,
+		LastClientID:   c.ID(),
+		LastGeneration: o.GetGeneration(),
+		StoppedCh:      stoppedCh,
+	}
+	// launch the goroutine to watch events
+	logger.V(consts.LogLevelDebug).Info("Starting event watcher", "meta", updatedMeta)
+	r.eventWatcherRegistry.Register(name, updatedMeta)
+	go r.getEvents(watchCtx, o, wsClient, stoppedCh)
```

Comment:
Are okay to skip the error evaluation here?

Reply:
I think so. It wasn't actually returning any errors so I removed that return in 2b28b51.